src/org/rascalmpl/values/parsetrees/TreeAdapter.java

```diff
@@ -139,6 +140,9 @@ public static IConstructor getProduction(ITree tree) {
 	/**
 	 * This function assumes that getLabeledField does not return null for the same parameters!
 	 */
+	// TODO @PieterOlivier I guess we could extend this the way we also extended the getLabeledField.
+	// if the field is still on the parsed side of the dot then we're fine. Otherwise we could
+	// think about shifting the dot if we put in correct trees at the right places?
```

Comment:
I think so. We could even remove the error node altogether when the dot reaches the end of the production.

Reply:
I see. Can that really happen? That the dot is already beyond the last one, but there is still an error on this level? My intuition would be that the error would always be higher up in the tree then, shifting its own dot over the child it has just recognized.

Reply:
That is not what I meant. If the dot is before the last child and you set that child with `putLabeldField`, the dot would shift to after the last child. That is the moment the error node can be removed altogether.

Reply:
o right! that makes sense.